components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtConfig.java

```diff
@@ -569,6 +569,7 @@ private void loadPolicyExtensions(Properties properties, String extensionType) {
             className = properties.getProperty(extensionType + "." + count);
             if (className == null) {
                 count++;
+                size--;
```

Comment:
There is a requirement for password policies to work for an example when following order of extensions are available [1],

Password.policy.extensions.1
Password.policy.extensions.4
Password.policy.extensions.6

In that case decrementing size won't satisfy this scenario.

[1]. https://wso2.org/jira/browse/IDENTITY-3202